modin/data_management/functions/groupby_function.py

```diff
@@ -278,16 +339,19 @@ def caller(
                     if isinstance(map_func, dict)
                     else map_func
                 )
-            return query_compiler.default_to_pandas(
-                lambda df: default_to_pandas_func(
-                    df.groupby(by=by, axis=axis, **groupby_args), **map_args
-                )
-            )
+
+            def groupby_reduce(df):
+                grp = df.groupby(by=by, axis=axis, **groupby_args)
+                if selection is not None:
+                    grp = grp[selection]
+                return default_to_pandas_func(grp, **map_args)
+
+            return query_compiler.default_to_pandas(groupby_reduce)
         assert axis == 0, "Can only groupby reduce with axis=0"
 
-        if numeric_only:
+        if numeric_only and selection is None:
```

Comment:
if columns are explicitly selected for aggregation we mustn't drop them as non-suitable for numeric aggregation

Reply:
done